test/unit/puppetlabs/puppetserver/lockable_pool_test.clj

```diff
@@ -508,17 +508,19 @@
           borrow-1 (.borrowItem pool)
           borrow-thread-started-2 (promise)
           borrow-thread-started-3? (promise)
+          lock-thread-locked? (promise)
           borrow-thread-2 (future
                            (deliver borrow-thread-started-2
                                     (Thread/currentThread))
+                           (timed-deref lock-thread-locked?)
                            (.borrowItem pool))
           borrow-thread-3 (future
                            (deliver borrow-thread-started-3? true)
+                           (timed-deref lock-thread-locked?)
                            (.borrowItem pool))
           borrow-thread-obj-2 @borrow-thread-started-2
           _ @borrow-thread-started-3?
           lock-thread-started? (promise)
-          lock-thread-locked? (promise)
           unlock? (promise)
           lock-thread (future
                        (deliver lock-thread-started? true)
```

Comment:
Yep, that's right.  They specifically are intending to validate what happens around borrow attempts that are made after the lock is already held.